src/components/developers/useDevelopers.ts

```diff
@@ -19,72 +19,91 @@ import useInfiniteScroll from '@/hooks/useInfiniteScroll';
 import CHANNEL_ID from '@/consts/channels';
 import { searchAll } from '@/api/search';
 
-//TODOapi 에러 처리, 1글자 이하이면 경고창이 필요한가?스낵바?, 페치중 페치 막기 , 초기 렌더링을 실행 막기
-const useDevelopers = () => {
-  const [isLoading, setIsLoading] = useState<boolean>(false);
-  const [isSearching, setIsSearching] = useState<boolean>(false);
+interface useDevelopersProps {
+  onGetFail: (error: unknown) => void;
+}
 
+const useDevelopers = ({ onGetFail }: useDevelopersProps) => {
   const dispatch = useAppDispatch();
-
   const developerList = useAppSelector(developerListSelector);
   const onlineUserList = useAppSelector(onlineUserListSelector);
   const headerSearchValue = useAppSelector(inputSelector);
-
-  const { page, pageEnd } = useInfiniteScroll({
-    options: { threshold: 0.2 },
-  });
+  const [isEndOfList, setIsEndOfList] = useState<boolean>(false);
+  const [isFetching, setIsFetching] = useState<boolean>(false);
+  const isLoading = useRef(false);
+  const setIsLoading = (state: boolean) => {
+    isLoading.current = state;
+  };
+  const { page, pageEnd } = useInfiniteScroll({ isFetching, options: {} });
```

Comment:
options 는 안넣어주셔도 됩니당~!

```suggestion
  const { page, pageEnd } = useInfiniteScroll({ isFetching });
```